tests/Benchmarks/System.Text.Primitives/GetUtf16CharCount.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using BenchmarkDotNet.Attributes;
+using System.Buffers.Text;
+using System.Collections.Generic;
+using System.IO;
+
+namespace System.Text.Primitives.Benchmarks
+{
+    public class GetUtf16CharCount
+    {
+        private const int InnerCount = 10000;
+
+        private static readonly UTF8Encoding _nonReplacingEncoder = new UTF8Encoding(encoderShouldEmitUTF8Identifier: false, throwOnInvalidBytes: true);
+        private static readonly byte[] _utf8BOM = new byte[] { 0xEF, 0xBB, 0xBF };
+
+        [ParamsSource(nameof(FileNames))]
+        public string FileName;
+
+        static byte[] utf8Text;
+
+        [GlobalSetup]
+        public void Setup()
+        {
```

Comment:
nit: you can use expressions for such simple methods

`public void Setup() => utf8Text = ReadTestResource(FileName).ToArray();`